docs/workers.md

```diff
@@ -66,23 +66,28 @@ https://hub.docker.com/r/matrixdotorg/synapse/.
 
 To make effective use of the workers, you will need to configure an HTTP
 reverse-proxy such as nginx or haproxy, which will direct incoming requests to
-the correct worker, or to the main synapse instance. See 
+the correct worker, or to the main synapse instance. See
 [reverse_proxy.md](reverse_proxy.md) for information on setting up a reverse
 proxy.
 
-To enable workers you should create a configuration file for each worker
-process. Each worker configuration file inherits the configuration of the shared
-homeserver configuration file.  You can then override configuration specific to
-that worker, e.g. the HTTP listener that it provides (if any); logging
-configuration; etc.  You should minimise the number of overrides though to
-maintain a usable config.
+When using workers, each worker process has its own configuration file which
+contains settings specific to that worker, such as the HTTP listener that it
+provides (if any), logging configuration, etc.
 
+Normally,  the worker processes are configured to read from a shared
+configuration file as well as the worker-specific configuration files. This
+makes it easier to keep common configuration settings synchronised across all the
+processes.
 
-### Shared Configuration
```

Comment:
could keep these headers?

Reply:
I'm not really sure how they interact with the prose though, since we explain the worker config files outside the worker configuration section. Which feels a bit weird to me?

Reply:
I think it's fine. the top section introduces the structure; the later sections explain the details of what you need to put in each file.

What I would say is that the "Worker configuration" section goes on to talk about starting synapse, which could do with its own `###`-level heading.